app/models/turbo/streams/tag_builder.rb

```diff
@@ -227,6 +227,8 @@ def action_all(name, targets, content = nil, allow_inferred_rendering: true, **r
   private
     def render_template(target, content = nil, allow_inferred_rendering: true, **rendering, &block)
       case
+      when content.respond_to?(:render_in)
+        content.render_in(@view_context, &block)
```

Comment:
This line could instead be `@view_context.render(content, &block)`, which would give more control to the view context but will be slightly slower because of the extra indirection.

We’re already coupled to the `render_in` method based on the `respond_to?` check in the previous line, though that too could be delegated to the view context if the view context exposes a renderable-checking method.

I personally think this coupling with the `render_in` interface is _fine_, but open to your input and wanted to highlight an alternative.

Reply:
My preference is to delegate to Action View as much as possible, going as far as invoking `render` when we know that it'll call `render_in` behind the scenes.

Reply:
That makes sense. The reason I didn’t do that was because we’re already coupled to `render_in` here via the `respond_to?` check. Shall I update it to this?

```ruby
when content.respond_to?(:render_in)
  @view_context.render(content, &block)
```